File: circle-constructor/circle.rb
class Circle
  attr_reader :radius

  def initialize(radius)
    @radius = radius
  end
end

class Circle
  attr_reader :radius

  def initialize(input)
    if input[:radius].nil?
      @radius = input[:diameter]/2
    else
      @radius = input[:radius]
    end
  end
end

class Circle
  attr_reader :radius

  def initialize(input)
    if input.kind_of?(Hash)
      if input[:radius].nil?
        @radius = input[:diameter] / 2
      else
        @radius = input[:radius]
      end
    else
      @radius = input
    end
  end
end


class Circle
  attr_reader :radius

  def initialize(options_or_radius)
    if options_or_radius.kind_of?(Hash)
      @radius = options_or_radius[:radius] || (options_or_radius[:diameter] / 2)
    else
      @radius = options_or_radius
    end
  end
end

File: geometric-classes-js/public/js/Circle.js
class Circle {
  constructor(radius, x, y) {
    this.radius = radius
    this.x = x || 0
    this.y = y || 0
  }

  diameter () {
    return this.radius * 2;
  }

  area() {
      return Math.PI * this.radius * this.radius;
    }

perimeter() {
    return 2 * Math.PI * this.radius;
  }
}
